overlay/lib/fetch.nix

```diff
@@ -34,22 +34,20 @@ rec {
   # to be downloaded, the whole index needs to be recloned to get the download URL, which is potentially expensive.
   fetchCrateAlternativeRegistryExpensive = { index, name, version, sha256 }: with buildPackages; stdenvNoCC.mkDerivation {
     name = "${name}-${version}.tar.gz";
+    src = builtins.fetchGit { url = index; ref = "master"; };
```

Comment:
In the original PR conversation:

https://github.com/tenx-tech/cargo2nix/pull/109#discussion_r421936521

If this suggestion is true, I would expect to see some treatment of the ssh env carried forward. Instead all mentions of ssh env are deleted.

An aside, the `GIT_SSH` being deleted below looks like a rather crazed hack;  it was originally getting environment variables into scope for build time use of `git` further down by failing a `fetchGit`?  :scream:

Reply:
There should be no need to load `$GIT_SSH` and `$SSH_AUTH_SOCK` explicitly under this new method because those were "crazed hacks" normalized by Nixpkgs and specific to the now-removed [`fetchgitPrivate`](https://github.com/NixOS/nixpkgs/blob/65e7d9c8c94d934c494ab4486992112db6237dc5/pkgs/build-support/fetchgit/private.nix). `$GIT_SSH` used to point to the Nix store path of the `ssh` binary from `openssh`, while `$SSH_AUTH_SOCK` used to be defined in terms of the `ssh-auth-sock=` setting specified by the user in their `$NIX_PATH`.

With `builtins.fetchGit`, this is no longer necessary. Because this command runs at evaluation-time rather than at build-time, SSH credentials required for private repos are automatically inherited from the environment without needing any such hacks.

This also means that we can also remove the `socat`hacks in CI (and possibly the macOS `nix-docker` remote builder), since all the authorization and cloning of private repositories happens outside the Nix builder.

Reply:
> If this suggestion is true, I would expect to see some treatment of the ssh env carried forward. Instead all mentions of ssh env are deleted.

Sounds like it wasn't true.  I buy all of this.  It's how I understood it.

Reply:
Indeed it wasn't. I experimented a bit locally, and I also checked the original `fetchgitPrivate` source, and found it wasn't the case. 😄